### PR TITLE
Enable overriding environ values

### PR DESCRIPTION
**Pull request by rastytheamateur**
Django's `override_settings` does not override values taken from `os.environ`.

After a discussion we agreed with @ziima that the best way to fix this is to temporarily rename affected keys in `os.environ`. The `setting_changed` signal is received when `settings_override` is called and `settings_override` should be used just and only in tests, so this solution should have no impact on usual use.

### CODE REVIEW

File: tests/test_appsettings.py [hunk 1]
```diff
@@ -773,6 +774,70 @@ class AppConf(appsettings.AppSettings):
...
+        assert appconf.three == "Env_3"
+
+        assert "FOUR" not in os.environ
+        assert "__DAP_FOUR" not in os.environ
+        assert appconf.four == "Def_4"
+
+    @mock.patch.dict(os.environ, {"SETTING": "ONE=Env_1 TWO=Env_2"})
+    def test_environ_nested_setting_invalidation(self):
+        class AppConf(appsettings.AppSettings):
+            setting = cast(Dict[str, str], appsettings.NestedDictSetting(
```
Comment (ziima) [inline tests/test_appsettings.py:822]: Is the cast necessary?

File: src/appsettings/__init__.py [hunk 2]
```diff
@@ -197,6 +200,36 @@ def check(cls):
...
+                    os.environ[setting_override_key] = os.environ[setting]
+                    del os.environ[setting]
+                elif not enter and setting_override_key in os.environ:
+                    os.environ[setting] = os.environ[setting_override_key]
+                    del os.environ[setting_override_key]
+                break
+
+    def invalidate_cache(self, *, signal=None, sender=None, setting=None, value=None, enter=None):
+        """
+        Invalidate cache and environ values. Run when receive ``setting_changed`` signal.
```
[the remark this answers is not in the document]
Comment (rastytheamateur) [reply src/appsettings/__init__.py]: Fixed.

File: src/appsettings/__init__.py [hunk 3]
```diff
@@ -197,6 +200,36 @@ def check(cls):
         if exceptions:
             raise ImproperlyConfigured("\n".join(exceptions))
 
-    def invalidate_cache(self, **kwargs):
-        """Invalidate cache. Run when receive ``setting_changed`` signal."""
+    def _manage_environ_invalidation(self, setting, enter):
+        """Manage keys and values in ``os.environ`` on setting change."""
+        for item in self.settings.keys():
+            if self.settings[item].full_name == setting:
+                setting_override_key = "{}{}".format(self.OS_ENVIRON_OVERRIDE_PREFIX, setting)
```
[the remark this answers is not in the document]
Comment (rastytheamateur) [reply src/appsettings/__init__.py]: Fixed.

File: tests/test_appsettings.py (hunk 1 above)
Comment (rastytheamateur) [reply tests/test_appsettings.py]: `mypy` throws some errors thet `NestedDictSetting` is not indexable

File: src/appsettings/__init__.py [hunk 4]
```diff
@@ -197,6 +201,34 @@ def check(cls):
         if exceptions:
             raise ImproperlyConfigured("\n".join(exceptions))
 
+    def manage_environ_invalidation(self, signal, sender, setting, value, enter):
```
Comment (ziima) [inline src/appsettings/__init__.py:204]: Signal handler needs `**kwargs` according to documentation. That would also narrow down the unused arguments.

File: tests/test_appsettings.py (hunk 1 above)
Comment (ziima) [reply tests/test_appsettings.py]: That's not very nice. We should resolve that somehow.

Comment (ziima) [reply tests/test_appsettings.py]: Created #89

File: src/appsettings/__init__.py (hunk 4 above)
Comment (rastytheamateur) [reply src/appsettings/__init__.py]: Fixed.

**Pull request merged**